ext/RMagick/rmilist.c

```diff
@@ -504,13 +505,23 @@ ImageList_montage(VALUE self)
 
     images = images_from_imagelist(self);
 
-    // If app specified a non-default composition operator, use it for all images.
-    if (montage->compose != UndefinedCompositeOp)
+    for (image = images; image; image = GetNextImageInList(image))
     {
-        Image *i;
-        for (i = images; i; i = GetNextImageInList(i))
+        if (montage->compose != UndefinedCompositeOp)
+        {
+            image->compose = montage->compose;
+        }
+        if (montage->change.background_color)
```

Comment:
What is the difference between `->` and `.` here?

Reply:
it use `->` when a member of a structure is a pointer variable.